elytron/src/test/java/org/wildfly/extension/elytron/SyslogAuditLogTestCase.java

```diff
@@ -0,0 +1,327 @@
+
+    public SyslogAuditLogTestCase() {
+        super(ElytronExtension.SUBSYSTEM_NAME, new ElytronExtension());
+    }
+
+    /**
+     * Creates the kernel services and sets up the base operations before each test
+     */
+    @Before
+    public void init() throws Exception {
+        services = super.createKernelServicesBuilder(new TestEnvironment()).setSubsystemXmlResource("syslog-audit-logging.xml").build();
+        if (!services.isSuccessfulBoot()) {
+            Assert.fail(services.getBootError().toString());
+        }
+        udpOperation = createUdpSyslogOperation(SERVER_ADDRESS);
+        badHostUdpOperation = createUdpSyslogOperation(BAD_SERVER_ADDRESS);
+    }
+
+    /**
+     * Initializes the syslog server for all tests
+     */
+    @BeforeClass
+    public static void initServer() throws Exception {
+        udpServer = SimpleSyslogServer.createUdp(UDP_PORT);
+    }
+
+    /**
+     * Resets the operations after each test
+     */
+    @After
+    public void shutdownServer() {
+        udpOperation = null;
+        badHostUdpOperation = null;
+    }
+
+    /**
+     * Closes the syslog server after all tests
+     */
+    @AfterClass
+    public static void closeServer() {
+        udpServer.close();
+    }
+
+    /**
+     * Sets the syslog format to RFC3164 and verifies the Elytron AuditEndpoint is created with RFC3164
+     * by verifying the connection message
+     */
+    @Test
+    public void testRfc3164SyslogFormat() throws Exception {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(RFC3164_STRING);
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC3164_SYSLOG_MESSAGE);
+    }
+
+
+    /**
+     * Sets the syslog format to RFC5424 and verifies the Elytron AuditEndpoint is created with RFC5424
+     * by verifying the connection message
+     */
+    @Test
+    public void testRfc5424SyslogFormat() throws Exception {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(RFC5424_STRING);
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+
+    /**
+     * Doesn't set a syslog and verifies the Elytron AuditEndpoint is created with the default value
+     * of RFC5424 by verifying the connection message
+     */
+    @Test
+    public void testDefaultSyslogFormat() throws Exception {
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the syslog configuration operation fails if an invalid RFC format is used
+     */
+    @Test
+    public void testWrongSyslogFormat() {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(BAD_RFC_STRING);
+        ModelNode response = services.executeOperation(udpOperation);
+        assertCorrectError(response, "No enum constant org.jboss.logmanager.handlers.SyslogHandler.SyslogType.RFC1");
```

Comment:
We shouldn't use the actual message here. We should use the message ID or a pattern of the message ID and the "RFC1" portion. The reason is this test could break if once the translations make it in.

Reply:
Updated